chain/epoch_manager/src/lib.rs

```diff
@@ -1105,25 +1113,25 @@ impl EpochManager {
             epoch_change = true;
         }
         let epoch_info = self.get_epoch_info(epoch_id)?.clone();
-        let mut proposals: HashMap<AccountId, ValidatorStake> = HashMap::default();
+        let mut new_aggregator = EpochInfoAggregator::new(epoch_id.clone(), *last_block_hash);
         let mut cur_hash = *last_block_hash;
+        let mut overwrite = false;
         while cur_hash != aggregator.last_block_hash || epoch_change {
             // Avoid cloning
             let prev_hash = self.get_block_info(&cur_hash)?.prev_hash;
             let prev_height = self.get_block_info(&prev_hash)?.height;
             let block_info = self.get_block_info(&cur_hash)?;
             if &block_info.epoch_id != epoch_id || block_info.prev_hash == CryptoHash::default() {
+                // This means that we reached the previous epoch and still hasn't seen
+                // `aggregator.last_block_hash` and therefore implies that a fork has happened.
```

Comment:
It doesn't necessarily imply that the fork has happened. If `epoch_start = true`, we are expected to get to this branch even if there was no forks.